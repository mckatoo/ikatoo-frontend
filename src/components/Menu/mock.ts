import { SideMenuProps } from '.'

export const mockedMenu: SideMenuProps = {
  links: [
    {
      label: 'Sobre',
      to: 'about'
    },
    {
      label: 'Habilidades',
      to: 'skills'
    },
    {
      label: 'Projetos',
      to: 'projects'
    },
    {
      label: 'Contato',
      to: 'contact'
    }
  ],
  social: [
    {
      name: 'github',
      url: 'https://github.com/mckatoo',
      url_icon: '/images/github.svg'
    },
    {
      name: 'youtube',
      url: 'https://youtube.com/mckatoo',
      url_icon: '/images/youtube.svg'
    },
    {
      name: 'linkedin',
      url: 'https://linkedin.com/in/mckatoo',
      url_icon: '/images/linkedin.svg'
    }
  ]
}
