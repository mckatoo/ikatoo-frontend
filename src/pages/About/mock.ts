import { AboutProps } from '.'

export const about: AboutProps = {
  skills: [
    'javascript',
    'dotnet',
    'nodedotjs',
    'html5',
    'css3',
    'typescript',
    'jest',
    'json',
    'git',
    'bootstrap',
    'react',
    'mysql',
    'npm',
    'php',
    'java'
  ],
  title: 'Ol√°. Bem vindo‚ùó',
  description:
    '<p>Me chamo Milton Carlos Katoo, moro em Itapira, interior de S√£o Paulo/Brasil. Pai de uma princesa e filho de excelente cozinheira Italiana e um saldoso Japon√™s faz tudo, sou um desenvolvedor full-stack que ama programa√ß√£o e desenvolvimento de software afim de melhorar a vida das pessoas.</p><p>Pessoa bem organizada, solucionador de problemas, funcion√°rio independente com alta aten√ß√£o aos detalhes.F√£ de animes, mangas, games, s√©ries de TV e filmes. Uma pessoa de fam√≠lia e pai de uma princesa.</p><p>Interessado em todo o espectro de programa√ß√£o e trabalhar em projetos ambiciosos com pessoas positivas.</p><a class="text-mck_aqua underline underline-offset-8" href="https://ikatoo.com.br/contact/" rel="contact"><span>üéâ</span>Vamos fazer algo especial.</a><span>üòÑ</span>'
}
