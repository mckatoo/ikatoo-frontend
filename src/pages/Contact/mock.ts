export default {
  title: 'Entre em contato ğŸ˜„',
  description: `Estou interessado em oportunidades CLT, mas nÃ£o descarto nenhum tipo de trabalho desde que me traga oportunidades de aprendizado. No entanto, se vocÃª tiver outra solicitaÃ§Ã£o ou pergunta, nÃ£o hesite em enviar um email. <a href="mailto:mckatoo@gmail.com?Subject=Contato%20pelo%20site">mckatoo@gmail.com</a>`,
  localization: {
    lat: -22.428850083857423,
    lng: -46.830700405308185
  }
}
