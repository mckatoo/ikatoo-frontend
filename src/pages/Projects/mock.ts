import { ProjectProps } from '.'

export const mockProjects: ProjectProps[] = [
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  },
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  },
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  },
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  },
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  }
]
