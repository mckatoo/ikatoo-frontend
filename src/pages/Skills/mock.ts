import { SkillsProps } from '.'

export const mockSkillsPageData: SkillsProps = {
  title: 'Habilidades e Experiências',
  description: `<p>Trabalhei como programador em VBA usando o MS Access em 2003, logo após formado no curso técnico do SENAC. Não dei muita sorte pois a empresa fechou 3 meses após eu entrar.</p>
  <p>Precisando de emprego mudei de área, fui trabalhar em uma grande indústria farmacêutica da cidade e lá fiquei durante 7 anos trabalhando como operador de máquinas na produção de medicamentos injetáveis, mas sempre com um pézinho na programação. Lá eu criei um programa em Delphi 7 com banco em MS Access para controle de produção de medicamentos liofilizados que utilizávamos para melhorar o fluxo de produção do setor e diminuir o tempo nos processos de fabricação.</p>
  <p>Sempre estudando banco de dados e programação, em 2013 surgiu uma oportunidade na faculdade onde trabalhei e estudei até julho de 2021.</p>
  <p>Lá atuei principalmente na manutenção e implementação de rede de computadores com 3 laboratórios de informática, departamentos administrativos e outros departamentos academicos. Implementei uma revista eletronica "Consciesi" e desenvolvi pequenas ferramentas para auxiliar o sistema administrativo e acadêmico.</p>
  <p>Atualmente estou desempregado procurando novas oportunidades.</p>
  <p>Visite meu perfil no <a target="_blank" href="https://linkedin.com/in/mckatoo">LinkedIn</a> para mais detalhes ou entre em <a href="/contact">contato</a> comigo.`,
  skills: [
    { skillTitle: 'Back-end', rankPercent: 40 },
    { skillTitle: 'Modelagem de dados', rankPercent: 30 },
    { skillTitle: 'Front-end', rankPercent: 20 },
    { skillTitle: 'CI/CD', rankPercent: 10 }
  ],
  lastJobs: [
    {
      jobTitle: 'Calm Organizador de Criptomoedas',
      jobDescription: 'Projeto pessoal para estudo.',
      yearMonthStart: '2022 - 03'
    },
    {
      jobTitle: 'Uniesi - Centro Universitário de Itapira',
      jobDescription:
        'Responsável pela infraestrutura local e suporte dos serviços dispostos pela mantenedora UNIP.',
      yearMonthStart: '2013 - 06',
      yearMonthEnd: '2021 - 07'
    },
    {
      jobTitle: 'Itacom Veículos',
      jobDescription:
        'Desenvolvimento do módulo venda de veículos do sistema de gerenciamento geral.',
      yearMonthStart: '2003 - 10',
      yearMonthEnd: '2003 - 12'
    }
  ]
}
