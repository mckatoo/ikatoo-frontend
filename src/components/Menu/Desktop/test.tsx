import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { describe, it } from 'vitest'

import DesktopMenu from '.'
import { mockedMenu } from '../mock'

describe('Menu Component', () => {
  it('should render menu on desktop', () => {
    const { getByText, getByLabelText, container } = render(
      <MemoryRouter initialEntries={[`/`]}>
        <DesktopMenu links={mockedMenu.links} social={mockedMenu.social} />
      </MemoryRouter>
    )

    expect(getByLabelText(/side menu/i)).toBeInTheDocument()
    expect(getByLabelText(/logotipo/i)).toBeInTheDocument()
    expect(getByText(/sobre/i)).toHaveClass('text-mck_aqua')
    expect(getByText(/habilidades/i)).toHaveClass('text-gray-500')
    expect(getByText(/habilidades/i)).not.toHaveClass('text-mck_aqua')
    expect(container).toMatchSnapshot()
  })
})
