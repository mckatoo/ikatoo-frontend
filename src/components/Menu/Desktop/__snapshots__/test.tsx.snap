// Vitest Snapshot v1

exports[`Menu Component > should render menu on desktop 1`] = `
.c1 svg {
  margin: auto;
}

.c0 {
  text-align: -webkit-center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 a {
  display: block;
  padding: 1rem 2.5rem;
  width: 100%;
}

.c3 {
  --tw-bg-opacity: 1;
  background-color: rgb(8 253 216 / var(--tw-bg-opacity));
}

.c3 a {
  --tw-text-opacity: 1;
  color: rgb(0 3 44 / var(--tw-text-opacity));
}

<div>
  <menu
    aria-label="Side Menu"
    class="text-center w-fit z-20 h-screen bg-slate-800 text-gray-500 flex flex-col justify-between"
  >
    <div
      class="c0 z-30 flex justify-center px-4 py-8 bg-mck_black"
    >
      <div
        aria-label="Logotipo"
        class="c1 text-gray-500 w-fit h-fit flex flex-col items-center text-center"
      >
        <a
          href="/"
        >
          <svg
            fill="none"
            viewBox="0 0 40 17"
            width="120"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M39.416.944H.3v15.3h39.116V.944Z"
              stroke=""
              stroke-width=".028"
            />
            <path
              d="M5.34 3.838v9.512h2.232V9.86h.02l.214-.21 3.234 3.666h2.703L9.257 8.23l4.486-4.392h-2.777L7.57 7.16V3.838h-2.23Z"
              fill="#FF2A2A"
            />
            <path
              d="M3.842 6.664v5.787H2.328V7.827c.308-1.11.939-1.036 1.514-1.163Zm0-2.817v1.768H2.328c-.017-1.98-.379-1.768 1.514-1.768ZM16.064 9.733c-.565 0-.96.073-1.184.217-.224.145-.336.393-.336.745 0 .261.078.47.233.625.155.155.365.232.63.232.4 0 .71-.15.93-.45.221-.302.331-.724.331-1.265v-.104h-.604Zm2.108-.584v3.303h-1.504v-.646a1.824 1.824 0 0 1-.702.589 2.15 2.15 0 0 1-.946.206c-.658 0-1.171-.174-1.54-.522-.365-.348-.548-.833-.548-1.457 0-.675.219-1.173.656-1.493.438-.324 1.115-.486 2.031-.486h1.05V8.39c0-.245-.09-.43-.27-.553-.175-.128-.435-.191-.78-.191-.361 0-.713.046-1.054.14a4.85 4.85 0 0 0-1.06.433V6.927c.335-.137.674-.24 1.019-.305.344-.065.71-.098 1.095-.098.94 0 1.6.191 1.98.574.382.382.573 1.066.573 2.051ZM22.384 5.02v1.644h1.979v1.163h-1.98v2.738c0 .259.063.445.187.558.127.11.337.166.63.166h1.163v1.162h-1.271c-.868 0-1.456-.137-1.762-.413-.307-.279-.46-.794-.46-1.545V7.827h-1.478V6.664h1.478V5.55c.074-.53 1.514-.53 1.514-.53ZM28.373 7.754c-.362 0-.646.16-.853.48-.206.318-.31.76-.31 1.329 0 .568.104 1.013.31 1.333.207.317.491.475.853.475.365 0 .651-.158.858-.475.206-.32.31-.765.31-1.333 0-.569-.104-1.011-.31-1.328-.207-.32-.493-.48-.858-.48Zm-2.677 1.809c0-.934.241-1.673.724-2.217.485-.548 1.136-.822 1.953-.822.82 0 1.471.274 1.953.822.486.544.729 1.283.729 2.217 0 .933-.243 1.674-.729 2.222-.482.544-1.133.816-1.953.816-.817 0-1.468-.272-1.953-.816-.483-.548-.724-1.289-.724-2.222ZM34.745 7.754c-.362 0-.646.16-.853.48-.207.318-.31.76-.31 1.329 0 .568.103 1.013.31 1.333.207.317.49.475.853.475.365 0 .65-.158.858-.475.206-.32.31-.765.31-1.333 0-.569-.104-1.011-.31-1.328-.207-.32-.493-.48-.858-.48Zm-2.677 1.809c0-.934.241-1.673.723-2.217.486-.548 1.137-.822 1.954-.822.82 0 1.47.274 1.953.822.486.544.729 1.283.729 2.217 0 .933-.243 1.674-.729 2.222-.482.544-1.133.816-1.953.816-.817 0-1.468-.272-1.954-.816-.482-.548-.723-1.289-.723-2.222Z"
              fill="#B3B3B3"
            />
          </svg>
          <p
            class="font-semibold"
          >
            Milton Carlos Katoo
          </p>
          <span
            class="text-sm text-gray-500"
          >
            Software Developer
          </span>
        </a>
      </div>
    </div>
    <nav
      class="text-lg"
    >
      <ul
        class="list-reset border-slate-700 border-y-2 divide-y divide-slate-700 font-semibold"
      >
        <li
          class="c2"
        >
          <a
            class="text-mck_aqua"
            href="/about"
          >
            Sobre
          </a>
        </li>
        <li
          class="c2"
        >
          <a
            class="text-gray-500"
            href="/skills"
          >
            Habilidades
          </a>
        </li>
        <li
          class="c2"
        >
          <a
            class="text-gray-500"
            href="/projects"
          >
            Projetos
          </a>
        </li>
        <li
          class="c2"
        >
          <a
            class="text-gray-500"
            href="/contact"
          >
            Contato
          </a>
        </li>
      </ul>
      <li
        class="c2 c3"
      >
        <a
          href="/login"
        >
          Entrar
        </a>
      </li>
    </nav>
    <div
      class="flex flex-row justify-around py-2"
    >
      <a
        class="w-6 text-inherit"
        href="https://github.com/mckatoo"
        rel="noreferrer"
        target="_parent"
      />
      <a
        class="w-6 text-inherit"
        href="https://youtube.com/mckatoo"
        rel="noreferrer"
        target="_parent"
      />
      <a
        class="w-6 text-inherit"
        href="https://linkedin.com/in/mckatoo"
        rel="noreferrer"
        target="_parent"
      />
    </div>
  </menu>
</div>
`
